specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2021-01-28-preview/privateLinkScopes.json

```diff
@@ -0,0 +1,1129 @@
+        "operationId": "PrivateEndpointConnections_ListByPrivateLinkScope",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/PrivateLinkScopeName"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successfully retrieved private endpoint connections.",
+            "schema": {
+              "$ref": "#/definitions/PrivateEndpointConnectionListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponseV2"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "Gets list of private endpoint connections on a private link scope.": {
+            "$ref": "./examples/PrivateEndpointConnectionList.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.HybridCompute/locations/{location}/privateLinkScopes/{privateLinkScopeId}": {
+      "get": {
+        "description": "Returns a Azure Arc PrivateLinkScope's validation details.",
+        "operationId": "PrivateLinkScopes_GetValidationDetails",
+        "parameters": [
+          {
+            "$ref": "#/parameters/LocationParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/PrivateLinkScopeId"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Azure Arc PrivateLinkScope validation details definition.",
+            "schema": {
+              "$ref": "#/definitions/PrivateLinkScopeValidationDetails"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponseV2"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "PrivateLinkScopeGet": {
+            "$ref": "./examples/PrivateLinkScopesGetValidation.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.HybridCompute/machines/{machineName}/privateLinkScope": {
```

Comment:
>privateLinkScope [](start = 129, length = 16)

The name of the `privateLinkScope` segment should be plural. Also, the GET on the `privateLinkScope` should be returning a collection of resources and not a single one. If you only have one you should use the singleton pattern where you implement list on /privateLinkScopes and GET on /privateLinkScopes/default. Other valid singleton names include latest, main, current.

Reply:
our scenario is that a machine resource can have one private link scope associated with it. this api is intended to return the current private link scope associated with this machine resource. would:

/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.HybridCompute/machines/{machineName}/privateLinkScopes/current

be what you're looking for?

I also don't understand what you mean by the GET on the privateLinkScope should be returning a collection. Above, where we're doing a GET across either the RG or the Sub, it does return a list. Here it should only return a single result because there is only ever one result.

Reply:
as a follow up, there was a request to have this api support also returning the machine data in the same response, controlled by a query parameter. Would the proposed route be acceptable for that?

Reply:
Basically, ARM resource IDs are pairs of segments with plural type followed by the identifier of that type. Let's take an example of Compute: `GET /subscriptions/{id}/resourceGroups/{rg}/providers/Microsoft.Compute/virtualMachines/{vm}/networkInterfaces` is used to return a collection of network interfaces on the VM. However, `POST /subscriptions/{id}/resourceGroups/{rg}/providers/Microsoft.Compute/virtualMachines/{vm}/restart` is used to restart a particular VM. The `networkInterfaces` and `restart` are in the same segment position and can mean either "collection" or "action" depending on the verb used.

In the case of your path `/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.HybridCompute/machines/{machineName}/privateLinkScope`, the `privateLinkScope` should be either a collection (hence the plural requirement) or an action. Given the GET verb, it's a collection.

To answer your question, I'm looking for the following operations:
- GET /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.HybridCompute/machines/{machineName}/privateLinkScopes
- GET /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.HybridCompute/machines/{machineName}/privateLinkScopes/current

Regarding the machine data on the same request, it's probably ok depending on what the payload looks like. Do you have an example you can paste into the comment.

Reply:
That all makes sense. Though the first GET without the /current would be largely useless in our case, as there can only ever be at most one. I hope we're not going to be required to have an api that returns a collection that will always contain 1 or 0 items.

Here's an example of the intended response from a test case. Without the query parameter requesting the machine data, we would omit the machineData

```
{
	"machineData": {
		"id": "/subscriptions/f930de4c-330f-45ea-b815-d52dada1718f/resourceGroups/RG/providers/Microsoft.HybridCompute/machines/Name",
		"name": "Name",
		"location": "westus",
		"tags": {
			"additionalProp1": "string",
			"additionalProp2": "string",
			"additionalProp3": "string"
		},
		"identity": {
			"type": "SystemAssigned",
			"principalId": "principalId",
			"tenantId": "tenantId"
		},
		"type": "Microsoft.HybridCompute/machines",
		"properties": {
			"provisioningState": "Succeeded",
			"agentVersion": "agent version",
			"status": "Connected",
			"lastStatusChange": "0001-01-01T00:00:00",
			"errorDetails": [],
			"displayName": "displayName",
			"machineFqdn": "machineFqdn",
			"osName": "Windows",
			"osVersion": "10",
			"osProfile": {
				"computerName": "computeName"
			},
			"vmId": "2cd2fbf5-feaf-4572-89ba-31fe8659e7b2",
			"vmUuid": null,
			"clientPublicKey": "clientPublicKey",
			"locationData": {
				"name": "name",
				"city": "city",
				"district": "district",
				"countryOrRegion": "country"
			},
			"osSku": null,
			"domainName": null,
			"adFqdn": null,
			"dnsFqdn": null,
			"privateLinkScopeId": null,
			"privateLinkScopeResourceId": "/subscriptions/f930de4c-330f-45ea-b815-d52dada1718f/resourceGroups/RG/providers/Microsoft.HybridCompute/privateLinkScopes/scopeName"
		}
	},
	"privateLinkScopeValidationData": {
		"id": "/subscriptions/f930de4c-330f-45ea-b815-d52dada1718f/resourceGroups/RG/providers/Microsoft.HybridCompute/privateLinkScopes/scopeName",
		"publicNetworkAccess": "Disabled",
		"connectionDetails": [
			{
				"id": "connectionDetailId",
				"privateIpAddress": "privateIp",
				"linkIdentifier": "linkId",
				"groupId": "groupId",
				"memberName": "memberName"
			}
		]
	}
}
```

Reply:
Since there's no truly standard naming for the "current" segment, how would the user know what it's called to do a get when exploring an API?

Reply:
this api is only intended for partner RPs to call, which would have the correct path coded, and we'd be checking that the calls are coming through ARM from partner RP app ids.

but in any case, how would you suggest we set up an api that behaves the way we need it to?

Reply:
@majastrz to try to make life easier, I went through and created a new PR with fewer commits and an initial commit of the unchanged previous api. hopefully that will make it easier to review this going forward. It's at:
https://github.com/Azure/azure-rest-api-specs/pull/13007

I'm going to close this PR in favour of that one.